pygmo/_py_problems.py

```diff
@@ -265,3 +265,158 @@ def get_decorator(self, fname):
+            prob = null_problem()
+        if type(prob) == problem:
+            # If prob is a pygmo problem, we will make a copy
+            # and store it. The copy is to ensure consistent behaviour
+            # with the other meta problems and with the constructor
+            # from a UDP (which will end up making a deep copy of
+            # the input object).
+            self._problem = deepcopy(prob)
+        else:
+            # Otherwise, we attempt to create a problem from it. This will
+            # work if prob is an exposed C++ problem or a Python UDP.
+            self._problem = problem(prob)
+
+        minBound, maxBound = self._problem.get_bounds()
+
+        inner_dim = len(minBound)
+        self.full_dim = inner_dim
+        if len(maxBound) != inner_dim:
+            raise ValueError("Problem bounds inconsistent!")
+
+        if len(fixed_arguments) != inner_dim:
+            raise ValueError("Got " + str(len(fixed_arguments)) + " argument array for problem of dimension " + str(inner_dim))
+
+        if self._problem.get_nix() > 0:
+            raise ValueError("Mixed integer-problems not yet supported.")
+
+        self.minBound = []
+        self.maxBound = []
+
+        for i in range(inner_dim):
+            arg = fixed_arguments[i]
+
+            if arg is None:
+                # free variable
+                self.minBound.append(minBound[i])
+                self.maxBound.append(maxBound[i])
+            else:
+                # fixed variable
+                if not arg >= minBound[i]:
+                    raise ValueError("Fixed argument " + str(arg) + " violates min bound " + str(minBound[i]))
+                if not arg <= maxBound[i]:
+                    raise ValueError("Fixed argument " + str(arg) + " violates max bound " + str(maxBound[i]))
+
+        # converting to internal format
+        self.fixed_arguments = [elem for elem in fixed_arguments if elem is not None]
+        self.fixed_flags = [elem is not None for elem in fixed_arguments]
+
+        assert(len(self.minBound) + sum(self.fixed_flags) == inner_dim)
+
+    def get_bounds(self):
+        return (self.minBound, self.maxBound)
+
+    def get_nobj(self):
+        return self._problem.get_nobj()
+
+    def get_nec(self):
+        return self._problem.get_nec()
+
+    def get_nic(self):
+        return self._problem.get_nic()
+
+    def get_nc(self):
+        return self._problem.get_nc()
+
+    def get_nx(self):
+        return len(self.minBound)
+
+    def get_nix(self):
+        return 0
+
+    def fitness(self, x) -> List[float]:
+        return self._problem.fitness(self.get_full_x(x))
+
+    def has_batch_fitness(self):
+        return self._problem.has_batch_fitness()
+
+    def batch_fitness(self, dvs):
+        contiguous_x = []
+        if len(dvs) % self.get_nx() != 0:
+            raise ValueError("Expected multiple of " + str(self.get_nx()) + " but got " + str(len(dvs)))
+        num_dvs = int(len(dvs) / self.get_nx())
```

Comment:
Can't you use the integer division operator here ``//``, rather than going through a float -> int conversion?